### PR TITLE
Feature connect vm ssh

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
--------------------------------------------------
add new action for connecting to machine by ssh.
UltimateConnectBySshAction.java is the action handler for ultimate version
CommunityConnectBySshAction.java is the action handler for community version.

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java [hunk 1]
```diff
@@ -32,39 +38,70 @@ public class ConnectBySshAction {
     private static final String CMD_SSH = "ssh %s@%s";
     private static final String CMD_SSH_KEY_PAIR = "ssh %s@%s -i %s";
 
-    public static void connectBySsh(VirtualMachine vm, @Nonnull Project project) {
+    @AzureOperation(name = "vm.connect_virtual_machine_ssh_ultimate", params = "vm.name()", type = AzureOperation.Type.TASK)
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java]: should be `ACTION`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/CommunityConnectBySshAction.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.jetbrains.plugins.terminal.ShellTerminalWidget;
+import org.jetbrains.plugins.terminal.TerminalView;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+public class CommunityConnectBySshAction {
+    private static final String SSH_TERMINAL_TABLE_NAME = "SSH - %s";
+    private static final String CMD_SSH_KEY_PAIR = "ssh %s@%s -i %s";
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/CommunityConnectBySshAction.java]: no need to set cert path which will be used by ssh by default

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java [hunk 3]
```diff
@@ -32,39 +38,70 @@ public class ConnectBySshAction {
...
+        return Paths.get(System.getProperty("user.home"), ".ssh", "id_rsa").toString();
+    }
+
+    @Nonnull
+    private static SshConfig toSshConfig(VirtualMachine vm, String name) {
+        final SshConfig config = new SshConfig(true);
+        config.setCustomName(name);
+        config.setId(UUID.nameUUIDFromBytes(name.getBytes()).toString());
+        config.setUsername(vm.getAdminUserName());
+        config.setAuthType(AuthType.KEY_PAIR);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java]: use vm's own auth type.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java [hunk 4]
```diff
@@ -32,39 +38,70 @@ public class ConnectBySshAction {
...
                 }
             }));
         }, AzureTask.Modality.ANY);
     }
 
-    private static String defaultSshPrivateKeyPath() {
-        Path userHomePath = Paths.get(System.getProperty("user.home"));
-        Path filePath = Paths.get(userHomePath.toString(), ".ssh", "id_rsa");
-        return filePath.toString();
+    private static String getDefaultSshPrivateKeyPath() {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java]: can be deleted

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/UltimateConnectBySshAction.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        );
+    }
+
+    private static void connectToSshUnderProgress(final @NotNull Project project, final @NotNull SshTerminalCachingRunner runner, final @NotNull RemoteCredentials data) {
+        try {
+            runner.connect();
+            AppUIUtil.invokeLaterIfProjectAlive(project, () ->
+                    TerminalView.getInstance(project).createNewSession(runner));
+        } catch (final RemoteSdkException e) {
+            AppUIUtil.invokeLaterIfProjectAlive(project, () ->
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/UltimateConnectBySshAction.java]: we may user `AzureMessager` to show notification and provide show config window action in the notifiaction

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-vm-lib/src/main/java/com/microsoft/azure/toolkit/ide/vm/VirtualMachineActionsContributor.java [hunk 6]
```diff
@@ -36,6 +37,11 @@ public void registerActions(AzureActionManager am) {
             .enabled(s -> s instanceof VirtualMachine && ((VirtualMachine) s).getFormalStatus().isRunning());
         am.registerAction(ADD_SSH_CONFIG, new Action<>(ADD_SSH_CONFIG, addSshConfigView));
 
+        final ActionView.Builder connectSshView = new ActionView.Builder("Connect using SSH", AzureIcons.Action.CONSOLE.getIconPath())
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-vm-lib/src/main/java/com/microsoft/azure/toolkit/ide/vm/VirtualMachineActionsContributor.java]: Connect Using SSH.

Comment (ivywei0125) [reply Utils/azure-toolkit-ide-libs/azure-toolkit-ide-vm-lib/src/main/java/com/microsoft/azure/toolkit/ide/vm/VirtualMachineActionsContributor.java]: done

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/UltimateConnectBySshAction.java (hunk 5 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/UltimateConnectBySshAction.java]: done

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java (hunk 3 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java]: done

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/CommunityConnectBySshAction.java (hunk 2 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/CommunityConnectBySshAction.java]: done

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java (hunk 1 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/ConnectBySshAction.java]: done

**Pull request closed without merging**